internal/builder/embedded/util.go

```diff
@@ -72,11 +71,15 @@ func solveRequestWithContext(sessionID string, image string, opts *config.BuildO
 	return req, nil
 }
 
-func showProgress(ch chan *controlapi.StatusResponse, noConsole bool) error {
-	displayCh := make(chan *bkclient.SolveStatus)
+func displayProgress(ch chan *controlapi.StatusResponse, progressFunc func(chan *bkclient.SolveStatus) error) error {
```

Comment:
I had some real trouble doing single producer, multi consumer and at some point just sort of decided that it's probably not worth it as a feature in general. If we do want it, I'll need some help with a better design.